### DISCUSSION TITLE
Signal Power and RSSI

### DISCUSSION THREAD

**Discussion by SFSailor** [jvde-github/AIS-catcher#328] [Q&A] [answered]
Outputs from the program variously provide "signalpower" or RSSI. Are these just synonyms for the same value?
Secondly, how is the displayed value calculated? I had seen somewhere a formula with a log and a square root, but cannot find it.  My understanding is that my RTL-SDR outputs an RSSI value of +/-127.

**Comment by jvde-github** [comment, marked as answer]
It is the same. All the samples are normalized to -1 to 1 (so for RTL-SDR this means / 127). And at sample rate of 48K/seconds, it takes 10 * log10 of the the average of the sum of squares of the IQ samples.

**Reply by SFSailor** [reply]
Thanks for the answer!

So, if I take that value as an exponent for ten (10^RSSI) I'll get a value directly proportional to received signal strength then.... right? This will be helpful for me scaling signal strength by distance from receiver.

**Comment by jvde-github** [comment]
Yes, you get the sum of squares of the sample, but typically I thought taking the 10log of that is a better measure for comparison?

________________________________
Van: SFSailor ***@***.***>
Verzonden: vrijdag 22 november 2024 01:07
Aan: jvde-github/AIS-catcher ***@***.***>
CC: Jasper ***@***.***>; Comment ***@***.***>
Onderwerp: Re: [jvde-github/AIS-catcher] Signal Power and RSSI (Discussion #328)


Thanks for the answer!

So, if I take that value as an exponent for ten (10^RSSI) I'll get a value directly proportional to received signal strength then.... right? This will be helpful for me scaling signal strength by distance from receiver.

**Reply by SFSailor** [reply]
The 10Log is just fine, and depending on how I calculate raw masthead signal strength may be quite convenient.  If not, I'll do other math.

Basically, the intent is to infer the radiated power at the ship's antenna based on RSSI and distance from receiver.  I'd expect to further scale the value for atmospherics based on a known good fixed transmitter operated by the USCG.  I have not particularly looked at coding the math yet.  First step is capturing the data so I can manipulate it.  This is not my strong suit!

**Reply by jvde-github** [reply]
Nice! At some point i was planning to use these RSSI and drift to detect spoofing (or just incorrect lat/lon). That is the nice thing about using the RTL-SDR. Never had time to look into it though. Let me know how it goes.

**Reply by el-eternauta** [reply]
Hi. Any further developments planned on this front? I see USCG and CBP MMSI's not transmitting position on my table of ships all the time. Would be nice to be able to "MLAT" them in some way with other stations.

**Comment by SFSailor** [comment]
Yeah, a pal and I actually proposed a similar anti-spoofing system to the
Coast Guard. We debated TDOA vs RSSI at some length.  My goal is actually
to detect poor installations on ocean-going boats, but with even two
receivers one could get a clue as to when a signal isn't quite kosher.

Of course, our local USCG has radar.....

**Comment by SFSailor** [comment]
I got interrupted with another related project, but I have several
receivers deployed and are feeding a database.  The real issue is whether
that RSSI data is useful.

What is MLAT?

**Reply by el-eternauta** [reply]
MLAT (or multilateratiom) is the analogue of this concept but as used in the ADS-B world for tracking planes which are transmitting in position-less Mode-S.

Basically it correlates the information from several overlapping receivers to calculate the airplane's position.

I'm new to AIS and it kinda surprised me that the same capability doesn't seem to be implemented.

**Comment by SFSailor** [comment]
Well, that is one application of getting the RSSI.  My original thought was
to use TDOA, but that's a bit more complex.

**Reply by el-eternauta** [reply]
Right. Besides needing to scale the value for atmospherics, wouldn't reported RSSI values potentially vary across stations because of things like type and length of run of coax, the presence or absence of an LNA, type of antenna (both polarization and impedance matching)...? Or am I misunderstanding the concept?

**Comment by SFSailor** [comment]
My intent is to scale all of them based on the RSSI of a known good strong
station.  Maybe that cancels out some of the factors.  For evaluating ship
installation quality, a statistical cloud of data will do the trick.  For
locating the transmitter, not so much,

A big factor is what way the boat is facing!

**Reply by el-eternauta** [reply]
Ah, yes.

that would work for me also. I'm in receiving range of a large, mountaintop, VTS station and several navigation aids...

**Comment by SFSailor** [comment]
The navaids seem the best.

**Comment by el-eternauta** [comment]
I used the navaids to calculate my compensation for frequency shift and now
average about 0.2 ppm per hour, which is fine I guess.

**Comment by SFSailor** [comment]
I have totally been ignoring frequency shift.  Is that gonna be a problem?

**Reply by jvde-github** [reply]
Nope. This is not an issue any more for newer dongles. More for older or cheaper dongles without frequency stabilization.

**Comment by el-eternauta** [comment]
Tbh, I have no idea.
